Source/WebKit/WebProcess/GPU/GPUProcessConnection.cpp

```diff
@@ -125,7 +125,7 @@ RefPtr<GPUProcessConnection> GPUProcessConnection::create(IPC::Connection& paren
     if (!connectionIdentifiers)
         return nullptr;
 
-    parentConnection.send(Messages::WebProcessProxy::CreateGPUProcessConnection(connectionIdentifiers->client, getGPUProcessConnectionParameters()), 0, IPC::SendOption::DispatchMessageEvenWhenWaitingForSyncReply);
+    parentConnection.send(IPC::createMessage<Messages::WebProcessProxy::CreateGPUProcessConnection>(WTFMove(connectionIdentifiers->client), getGPUProcessConnectionParameters()), 0, IPC::SendOption::DispatchMessageEvenWhenWaitingForSyncReply);
```

Comment:
This introduces a new,  arguably inferior send type unless needed. E.g. the rule would be:
- If sending by-copy, use `send(Messages::WebProcessProxy::CreateGPUProcessConnection { ... }, 0, IPC::SendOption::DispatchMessageEvenWhenWaitingForSyncReply);`
- If sending by-move, use `send(createMessage<Messages::WebProcessProxy::CreateGPUProcessConnection>( ...), 0, IPC::SendOption::DispatchMessageEvenWhenWaitingForSyncReply);`

This would be very unfortunate: laborous to keep consistent and frustrating to receive review nag about. Upon one seeing the `createMessage` function, uninformed developer would look up the definition only to learn rather impenetrable template (a message can be instance of `Message<MessageType>` or `MessageType`, but then Trait will be inherited from something and then some `m_arguments` are not being used in some struct but not in other struct...).

What I'm a concerned about is that most developers are mostly cut-pasting an existing send invocation for their new code. The two different flavours of sending messages is not a long-term benefit.

If we would end up with two different flavours of sending messages, I would see this being the requirement:
- The new, more expressive flavour would be all around superior, e.g. more or equally ergonomic to the old one
- It wouldn't matter if the developers would use the new, expressive form, as that would be the preferable anyway
- Over time, we would transition all call sites to the new form
(Example A below)

The alternative is to have only one, the current, form. This would mean that we would either define which messages receive the by-move values, or then we would hard-core the types that are always by-move, and then the copy call sites would explicitly copy (Example B below). 


Example A call site I could easily imagine to be ergonomic:
```
parentConnection.send<Messages::WebProcessProxy::CreateGPUProcessConnection>({ 0, IPC::SendOption::DispatchMessageEvenWhenWaitingForSyncReply}, WTFMove(connectionIdentifiers->client), getGPUProcessConnectionParameters());
```
E.g. we would pack the send metadata (destination, options, timeout) into the first parameter with initialiser constructor similar to `WebCore::ImagePaintingOptions`. The rest would be a template parameter pack that is in this patch stored (redundantly) to `Message<T>::m_arguments`.

So this we would implement only for `Connection` and `StreamClientConnection`, as we should probably remove `MessageSender` and later make stream connections symmetric.

Example B:

We would hard-code that types like `IPC::Connection::Handle`, `MachSendRight`, `Win32FileHandle` and `UnixFileDescriptor` are move-only. E.g.  `CreateGPUProcessConnection::m_arguments` would contain `IPC::Connection::Handle` instead of current `const IPC::Connection::Handle&`.

Moving call would be similar to today:
```
parentConnection.send(Messages::WebProcessProxy::CreateGPUProcessConnection(WTFMove(connectionIdentifiers->client), getGPUProcessConnectionParameters()), 0, IPC::SendOption::DispatchMessageEvenWhenWaitingForSyncReply);
```


I still want to stress:
I don't know any real use-case today where it would be useful to for call site X send type `T1` as _copy_, and then call site `Y` as _move_. This implies that currently we don't need the full expressivity of being able to pass lvalues and rvalues interchangeably. For current use-cases, selecting one is enough.

To me this implies that we should not take a big risk or big hit in maintainability, just for the sake of expressivity that is not needed in practice.

As a long term syntax for sending messages, to me Example A would be the preferable. If I understand correctly, the current syntax of generating structs that store the arguments and passing instances of the struct was dictated by older C++ shortcomings. Currently we have already done away with the reply struct instance, and I don't see any benefit of the send struct instance.